### PR TITLE
Implement `BacklogStore` via `RocksDB`

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

File: spectrum-offchain/src/backlog/persistence.rs [hunk 1]
```diff
@@ -9,7 +13,51 @@ where
...
+impl<TOrd> BacklogStore<TOrd> for RocksDBClient
+where
+    TOrd: OnChainOrder + Serialize + DeserializeOwned + Send + 'static,
+    TOrd::TOrderId: Serialize + DeserializeOwned + Send,
+{
+    async fn put(&mut self, ord: BacklogOrder<TOrd>) {
+        let db = self.db.clone();
+        spawn_blocking(move || {
+            db.put(
+                bincode::serialize(&ord.order.get_self_ref()).unwrap(),
```
Comment (kettlebell) [inline spectrum-offchain/src/backlog/persistence.rs:31]: The key that's used is simply the serialized representation of `TOrderId`. Do you want a particular prefix/postfix for the key?

File: spectrum-offchain/src/backlog.rs [hunk 2]
```diff
@@ -368,34 +372,70 @@ mod tests {
...
 
     #[tokio::test]
-    async fn should_suspend_existing_order() {
-        let mut backlog = setup_backlog(10, 5, 50);
+    async fn should_suspend_existing_order_mock_store() {
+        should_suspend_existing_order(MockBacklogStore::new()).await
+    }
+
+    #[tokio::test]
+    async fn should_suspend_existing_order_rocksdb() {
```
Comment (oskin1) [inline spectrum-offchain/src/backlog.rs:380]: What's the point in duplicating those tests depending on Store type? Those tests test Backlog logic which should not depend on BacklogStore implementation at all.

Comment (kettlebell) [reply spectrum-offchain/src/backlog.rs]: Wanted to transitively test the RocksDB implementation but I see your point. Removed. Adding separate test for RocksDB BacklogStore.

File: spectrum-offchain/src/backlog/persistence.rs (hunk 1 above)
Comment (oskin1) [reply spectrum-offchain/src/backlog/persistence.rs:31]: No need for that

**Review by oskin1 — APPROVED** [review]

**Pull request merged**